Final/CharCreator.tsx

```diff
@@ -0,0 +1,744 @@
+            
+        } catch (error) {
+            console.error('Error saving character data:', error);
+        }
+    
+        navigation.navigate('CharViewerStack', { 
+            screen: "CharViewer",
+            params: { character: newCharacter }
+        });
+    };
+
+    return (
+        <View style={styles.container}>
+            <TouchableOpacity
+            onPress={createCharacter}
+            style={styles.confirmButton}
+            >
+            <Text style={styles.buttonText}>Confirm</Text>
+            </TouchableOpacity>
+
+            <TextInput
+                placeholder="Enter Character Name"
+                value={character.name}
+                onChangeText={handleNameChange}
+                style={styles.input}
+            />
+
+            <Picker
+                selectedValue={selectedOption}
+                onValueChange={handleDropdownChange}
+                style={styles.dropdown}
+            >
+                <Picker.Item label="Race" value="race" />
+                <Picker.Item label="Class" value="class" />
+                <Picker.Item label="Background" value="background" />
+            </Picker>
+
+            {renderRaceSection()} 
+            {renderSubraceSection()}
+            {renderClassSection()}
+            {renderSubClassSection()}
+            {renderBackgroundSection()}
+            
+
+            {selectedOption && customizationSections[selectedOption]}
+        </View>
+    );
+}
+
+const styles = StyleSheet.create({
+    raceContainer: {
+    flexDirection: 'row',
+    },
+    raceColumn: {
+    flex: 1,
+    color: 'white',
+    },
+    subraceColumn: {
+    flex: 1,
+    color: 'white',
+    },
+    subraceContainer: {
+    flex: 1,
+    },
+    classContainer: {
+    flexDirection: 'row',
+    },
+    classColumn: {
+    flex: 1,
+    color: 'white',
+    },
+    subclassColumn: {
+    flex: 1,
+    color: 'white',
+    },
+    subclassContainer: {
+    flexDirection: 'row',
+    },
+    container: {
+    flex: 1,
+    backgroundColor: "black",
+    padding: 16,
+    },
+    confirmButton: {
+    backgroundColor: "red",
+    padding: 12,
+    borderRadius: 8,
+    alignItems: "center",
+    marginBottom: 16,
+    },
+    buttonText: {
+    color: "white",
+    fontSize: 18,
+    },
+    input: {
+    backgroundColor: "white",
+    padding: 12,
+    borderRadius: 8,
+    marginBottom: 16,
+    },
+    dropdown: {
+    backgroundColor: "gray",
+    color: "white",
+    borderRadius: 8,
+    height: 150,
+    },
+    sectionContainer: {
+    backgroundColor: "black",
+    padding: 16,
+    borderRadius: 8,
+    marginBottom: 16,
+    },
+    sectionTitle: {
+    color: "white",
+    fontSize: 16,
+    fontWeight: "bold",
+    marginBottom: 8,
+    },
+    radioButtonLabel: {
+    color: "white",
+    fontSize: 14,
+    marginBottom: 4,
+    },
+    sectionContainerStyle: {
+        backgroundColor: 'black',
+        padding: 16,
```

Comment:
using `flex:1` should solve the scrolling issue you demonstrated in class. A scroll view needs a parent container defined, so this parent `View` will define a fixed hieight.